// Copyright (c) 2023 Boston Dynamics, Inc.  All rights reserved.
//
// Downloading, reproducing, distributing or otherwise using the SDK Software
// is subject to the terms and conditions of the Boston Dynamics Software
// Development Kit License (20191101-BDSDK-SL).

syntax = "proto3";

package bosdyn.api;
option go_package = "bosdyn/api/directory_service";

option java_outer_classname = "DirectoryServiceProto";

import "bosdyn/api/directory.proto";

// DirectoryService lets clients discover which API services are available on a robot.
service DirectoryService {
    // Get information about a specific service.
    rpc GetServiceEntry(GetServiceEntryRequest) returns (GetServiceEntryResponse);

    // List all known services at time of call.
    rpc ListServiceEntries(ListServiceEntriesRequest) returns (ListServiceEntriesResponse);

}
